cardinal/types/engine/context.go

```diff
@@ -0,0 +1,36 @@
+package engine
+
+import (
+	"github.com/rs/zerolog"
+	"pkg.world.dev/world-engine/cardinal/gamestate"
+	"pkg.world.dev/world-engine/cardinal/receipt"
+	"pkg.world.dev/world-engine/cardinal/txpool"
+	"pkg.world.dev/world-engine/cardinal/types"
+	"pkg.world.dev/world-engine/sign"
+)
+
+type Context interface {
```

Comment:
A user  will end up looking at this interface when trying to figure out what they are allowed to do in a System and Query. The User appropriate methods should be placed at the top with comments, and the internal stuff should be at the bottom.